syntax = "proto3";

import "google/protobuf/empty.proto";

message SimpleMessage {
    string sender = 1;
    string message = 2;
    string timestamp = 3;
}

enum state {
    OK = 0;
    WARNING = 1;
    CRITICAL = 2;
}

message StateMessage {
    state status = 1;
    string message = 2;
}

service MessageService {
    rpc SendMessage(SimpleMessage) returns (StateMessage);
    rpc HealthCheck(google.protobuf.Empty) returns (StateMessage);
}
 